src/msg/producer/writer/consumer_writer.go

```diff
@@ -131,15 +169,32 @@ func newConsumerWriter(
 		opts = NewOptions()
 	}
 
-	var (
-		connOpts = opts.ConnectionOptions()
-		rw       = bufio.NewReadWriter(
-			bufio.NewReaderSize(u, connOpts.ReadBufferSize()),
-			bufio.NewWriterSize(u, connOpts.WriteBufferSize()),
-		)
-	)
+	connOpts := opts.ConnectionOptions()
+
+	var reader connReader
+	switch connOpts.CompressType() {
```

Comment:
it would be really nice in terms of ease of migration to actually prepend a header eg. {0xFF, 0xF0+compressType} and fall back to uncompressed en/decoders in case the header's wrong.